cli/cmd/simulation/clustercomplete.hpp

```diff
@@ -0,0 +1,207 @@
+            command(
+                e,
+                "ClusterComplete is a proof of concept of the more general idea of state space pruning in a cluster "
+                "hierarchy. In the application on SiDB layouts, it is able to simulate SiDB logic in base 3 for well "
+                "over 50 DBs, thereby redefining what was thought to be possible in the exact SiDB simulation domain.")
+    {
+        add_option("--epsilon_r,-e", physical_params.epsilon_r, "Electric permittivity of the substrate (unit-less)",
+                   true);
+        add_option("--lambda_tf,-l", physical_params.lambda_tf, "Thomas-Fermi screening distance (unit: nm)", true);
+        add_option("--mu_minus,-m", physical_params.mu_minus, "Energy transition level (0/-) (unit: eV)", true);
+        add_option("--global_potential,-g", params.global_potential,
+                   "Global potential applied to the entire layout (unit: V)", true);
+        add_option("--witness_partitioning_limit,-w", params.validity_witness_partitioning_max_cluster_size_gss,
+                   "The limit on the cluster size before Ground State Space omits the check for which it solves the "
+                   "validity witness partitioning NP-complete sub-problem; values above 15 severely impact the runtime",
+                   true);
+    }
+
+  protected:
+    /**
+     * Function to perform the simulation call.
+     */
+    void execute() override
+    {
+        // reset sim result
+        sim_result = {};
+        min_energy = std::numeric_limits<double>::infinity();
+
+        if (physical_params.epsilon_r <= 0)
+        {
+            env->out() << "[e] epsilon_r must be positive" << std::endl;
+            reset_params();
+            return;
+        }
+        if (physical_params.lambda_tf <= 0)
+        {
+            env->out() << "[e] lambda_tf must be positive" << std::endl;
+            reset_params();
+            return;
+        }
+
+        auto& s = store<fiction::cell_layout_t>();
+
+        // error case: empty cell layout store
+        if (s.empty())
+        {
+            env->out() << "[w] no cell layout in store" << std::endl;
+            reset_params();
+            return;
+        }
+
+        const auto get_name = [](auto&& lyt_ptr) -> std::string { return fiction::get_name(*lyt_ptr); };
+
+        const auto clustercomplete = [this, &get_name](auto&& lyt_ptr)
+        {
+            using Lyt = typename std::decay_t<decltype(lyt_ptr)>::element_type;
+
+            if constexpr (fiction::has_sidb_technology_v<Lyt>)
+            {
+                if constexpr (fiction::is_charge_distribution_surface_v<Lyt>)
+                {
+                    env->out() << fmt::format(
+                                      "[w] {} already possesses a charge distribution; no simulation is conducted",
+                                      get_name(lyt_ptr))
+                               << std::endl;
+                }
```

Comment:
```suggestion
                    env->out() << fmt::format(
                                      "[w] {} already possesses a charge distribution; no simulation is conducted\n",
                                      get_name(lyt_ptr));
                }
```
Please do this everywhere. Thank you!